### PR TITLE
feat: add interface files from `relayer-v2`

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR adds relayer typings from the `relayer-v2` repository into the SDK.

### CODE REVIEW

File: src/interfaces/Report.ts
Comment (pxrl) [inline src/interfaces/Report.ts:1]: This one seems exclusively used by the relayer "reporter" and it seems to be related to how it produces user-facing information. I'd be inclined to leave it in relayer-v2 until we find a reason for it to be in the SDK. wdyt?

**Review by pxrl — APPROVED** [review]
LGTM - I have open questions about a couple of files but the rest looks fine!

File: src/interfaces/Token.ts
[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/interfaces/Token.ts:1]: Good call! Removed

File: src/interfaces/Report.ts
Comment (james-a-morris) [reply src/interfaces/Report.ts:1]: Good call - remvoed

File: src/interfaces/SpokePool.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+export interface UnfilledDepositsForOriginChain {
+  [originChainIdPlusDepositId: string]: UnfilledDeposit[];
+}
+
+export interface Refund {
+  [refundAddress: string]: BigNumber;
+}
+export type FillsToRefund = {
```
[the remark this answers is not in the document]
Comment (dohaki) [reply src/interfaces/SpokePool.ts:115]: Or in general, how do you decide which one to use?

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/interfaces/SpokePool.ts:115]: This is the code directly from the `relayer-v2` that has been imported. My guess is that the `interface` type is used when there was a need to eventually extend/be extended by other interfaces.

**Pull request merged**